File: samples/sqlanywhere/externalenvironments/php/datatype.php
<?php
#   *******************************************************************
#   Copyright 2013 SAP AG or an SAP affiliate company. All rights reserved.
#   This sample code is provided AS IS, without warranty or liability of
#   any kind.
#  
#   You may use, reproduce, modify and distribute this sample code without
#   limitation, on the condition that you retain the foregoing copyright 
#   notice and disclaimer as to the original code.  
#   
#   *******************************************************************
?>

<?php
function fetch_bit( $in1 )
{
    return 1 - $in1;
}

function fetch_smallint( $in1 )
{
    return -16000 - $in1;
}

function fetch_usmallint( $in1 )
{
    return 32000 + $in1;
}

function fetch_int( $in1 )
{
    return -2000000000 - $in1;
}

function fetch_uint( $in1 )
{
    return 3000000000 + $in1;
}

function fetch_bigint( $in1 )
{
    return -17000000000000 - $in1;
}

function fetch_double( $in1 )
{
    return 3.14159 + $in1;
}

function fetch_string( $in1, $in2, $in3 )
{
    if( $in1 == 1 )
    {
	return $in2;
    }

    if( $in1 == 3 )
    {
        return $in2 . $in3;
    }

    return $in3;
}

?>

File: samples/sqlanywhere/externalenvironments/perl/datatype.pl
# ***************************************************************************
# Copyright (c) 2013 SAP AG or an SAP affiliate company. All rights reserved.
# ***************************************************************************
# This sample code is provided AS IS, without warranty or liability of
# any kind.
# 
# You may use, reproduce, modify and distribute this sample code without
# limitation, on the condition that you retain the foregoing copyright 
# notice and disclaimer as to the original code.  
# 
# *******************************************************************

sub fetch_bit
{
    #returns a bit
    $_[0] = 1 + $_[0];
}

sub fetch_smallint
{
    #returns a smallint
    $_[0] = -16000 - $_[0];
}

sub fetch_usmallint
{
    #returns an unsinged smallint
    $_[0] = 32000 + $_[0];
}

sub fetch_int
{
    #returns an integer
    $_[0] = -2000000000 - $_[0];
}

sub fetch_uint
{
    #returns an unsigned integer
    $_[0] = 3000000000 + $_[0];
}

sub fetch_string
{
    #takes an integer and two strings as input and returns the first
    #string if the integer input is 1, otherwise returns the second string
    if( $_[0] == 1 )
    {
	$_[1];
    }
    else
    {
	$_[2];
    }
}

sub fetch_outputs
{
    #returns 6 output variables of type bit, smallint, unsigned smallint,
    #integer, unsigned integer, string
    $_[0] = 0;
    $_[1] = -16020;
    $_[2] = 32300;
    $_[3] = -2000000001;
    $_[4] = 3009000000;
    $_[5] = "this is a string returned in an output variable";
}

